### PR TITLE
refactor Group flow and some fixes around User

### PR DESCRIPTION
**Pull request by atayarani**
@marcuswalser @Tapjoy/eng-group-ops 

Notes:
* Tested by creating/deleting/add_user for groups and testing create flow again
* Requires #3 to be merged first

### CODE REVIEW

File: lib/tapjoy/ldap/group/create.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            :cn          => opts[:name],
+            :objectclass => %w(top posixGroup),
+            :gidnumber   => Tapjoy::LDAP::client.get_max_id('group', opts[:type])
+          }
+        end
+      end
+    end
+  end
+end
+# def create
```
Comment (ehealy) [inline lib/tapjoy/ldap/group/create.rb]: If we no longer need this stuff we can clean it up.

File: lib/tapjoy/ldap/group/delete.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module Tapjoy
+  module LDAP
+    module Group
+      # Delete LDAP group
+      class Delete
+        def delete
+          puts Tapjoy::LDAP::client.delete(dn)
```
Comment (ehealy) [inline lib/tapjoy/ldap/group/delete.rb]: I'd think you'd want Trollop::die for name and group here.

File: lib/tapjoy/ldap/group/delete.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        end
+
+        private
+        def opts
+          @opts ||= Trollop::options do
+            # Set help message
+            usage 'group delete [options]'
+            synopsis "\nThis command is for deleting LDAP groups"
+
+            opt :group, 'Specify group', type: :string, required: true
```
Comment (ehealy) [inline lib/tapjoy/ldap/group/delete.rb]: need to parse :name

**Comment by atayarani** [comment]
@ehealy how does it look now?

**Pull request closed without merging**